crates/refunder/src/refund_service.rs

```diff
@@ -98,63 +93,55 @@ impl RefundService {
     async fn identify_uids_refunding_status_via_web3_calls(
         &self,
         refundable_order_uids: Vec<EthOrderPlacement>,
-    ) -> Result<Vec<OrderUid>> {
+    ) -> Vec<(OrderUid, Vec<CoWSwapEthFlowAddress>)> {
         let mut batch = Web3CallBatch::new(self.web3.transport().clone());
         let futures = refundable_order_uids
             .iter()
             .map(|eth_order_placement| {
                 let order_hash: [u8; 32] = eth_order_placement.uid.0[0..32]
                     .try_into()
                     .expect("order_uid slice with incorrect length");
-                let order = self
-                    .ethflow_contract
-                    .orders(ethcontract::tokens::Bytes(order_hash))
-                    .batch_call(&mut batch);
+
+                let contract_calls = self
+                    .ethflow_contracts
+                    .iter()
+                    .map(|contract| {
+                        contract
+                            .orders(ethcontract::tokens::Bytes(order_hash))
+                            .batch_call(&mut batch)
+                    })
+                    .collect::<Vec<_>>();
+
                 async move {
-                    let order_owner = match order.await {
-                        Ok(order) => Some(order.0),
-                        Err(err) => {
-                            tracing::error!(
-                                "Error while getting the currentonchain status of orderhash {:?}, \
-                                 {:?}",
-                                H256(order_hash),
-                                err
-                            );
-                            return None;
+                    // Theoretically, multiple orders with the same order_hash can exist in multiple
+                    // contracts at the same time. That's why we need to return a list of contracts
+                    // from which the order should be refunded.
```

Comment:
The entire order owner is part of the order_uid so since the ethflow contract is technically the order owner the owner bytes of the 2 order uids would be different for identical orders placed on multiple ethflow contracts.

Reply:
> contract is technically the order owner

I assumed the same but apparently this is not true for PreSign orders: 
https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/database/onchain_order_events/mod.rs#L642

Reply:
Now I see this: https://github.com/cowprotocol/ethflowcontract/blob/f466b7a8d5df80b593aeb05488e1c27afc7f2704/src/CoWSwapEthFlow.sol#L105
This means that all orders are 1271 orders, I guess, meaning I can leverage the fact that owner is always the EthFlowContract

Reply:
Resolved.